### ISSUE TITLE
Invent preset serialization

### ISSUE THREAD

**Issue by GitVex** [GitVex/BardicInspiration#99]
Usable for local storage (https://github.com/GitVex/BardicInspiration/issues/98) & presets (https://github.com/GitVex/BardicInspiration/issues/101)

**Comment by GitVex** [comment]
To serialize:
- video url
- volume
- timestamp
- skip-from skip-to state (https://github.com/GitVex/BardicInspiration/issues/100)
- equalizer settings (https://github.com/GitVex/BardicInspiration/issues/97)

**Comment by GitVex** [comment]
added preliminary preset interface

**Comment by GitVex** [comment]
Move the core presetState from `PlayerUI.tsx` to `PlayerHolderProvider.tsx`.
This intergrates the currently present initialPausedAtState and gives access to the context from the `ListItem.tsx` component

**Comment by GitVex** [comment]
Done ^

Left to do: Check all places where presetDispatch is needed. (ListItem Add method)
Ensure that the presetState is kept up to date (especially title, id and pausedAt)

**Comment by GitVex** [comment]
Massive performance issues when trying to continually update volume and other frequent updates.

Look into debouncing the dispatch calls.
A debounce function is in the utils/utils.ts file

```
let debouncedSetSaveDispatch: (value: { hasSaved: boolean; prevVol?: number }) => void | undefined;
	const savedVolume = player?.savedVolume ?? { hasSaved: false, prevVol: 0 };
	const setSavedVolume = (value: { hasSaved: boolean; prevVol?: number }) => {
		console.log('setting saved volume');

		if (!debouncedSetSaveDispatch) {
			debouncedSetSaveDispatch = debounce(() => {
				dispatch?.({
					type: 'setSavedVolume',
					index: playerId,
					payload: value,
				});
			}, 500);
		}

		debouncedSetSaveDispatch(value);		
	};
```

this (sort of) works but doesnt have the apropriate performance impact.

**Comment by GitVex** [comment]
https://youtu.be/gwIkg1acujU?si=hdbz3Mfu51qPN4HP

**Comment by GitVex** [comment]
performance issues resolved with 8117ad4db281af9d0f285fa151fac2c06664f981

Check if all parameters are appropriately updates in the preset State

**Comment by GitVex** [comment]
framePlayer doesnt need to be saved in the localStorage.

**Comment by GitVex** [comment]
Most presetState persistence done in aca149b65e24cf8bfa028dce0acfc350545a704b

Weird performance issue in vercel preview deploy

**Comment by GitVex** [comment]
merged into dev with ea3d6189fd1fe376b4f771182f1a966131f844ea

Still performance issues on prod?

**Comment by GitVex** [comment]
Issue resolved by switching from vercel hosting to self-hosted VPS deployment.

**Comment by GitVex** [comment]
Closing this for now, future additions to persistence should be proposed in new issues

**Comment by GitVex** [comment]
this also closes #101 and #98
